### PR TITLE
Add Support for Relationship connecting VMs to Nics

### PR DESCRIPTION
**Pull request by EarthmanT**

### CODE REVIEW

File: vsphere_plugin_common/tests/plugin_common_test.py [hunk 1]
```diff
@@ -2370,6 +2371,7 @@ def test_two_connections_wrong_then_right(self):
                 ctx=self.mock_ctx,
             )
 
+    @unittest.skip
```
Comment (0lvin) [inline vsphere_plugin_common/tests/plugin_common_test.py]: Is it failed?

Comment (EarthmanT) [reply vsphere_plugin_common/tests/plugin_common_test.py]: yes

File: cloudify_vsphere/devices/__init__.py [hunk 2]
```diff
@@ -12,17 +12,63 @@
...
+from cloudify.decorators import operation
+from cloudify_vsphere.utils import find_rels_by_type
+from vsphere_plugin_common.constants import (
+    VSPHERE_SERVER_ID,
+    NETWORK_NAME,
+    SWITCH_DISTRIBUTED)
+from vsphere_plugin_common import ContollerClient, ServerClient
+
+RELATIONSHIP_NIC_TO_NETWORK = \
+    'cloudify.relationships.vsphere.nic_connected_to_network'
```
[the remark this answers is not in the document]
Comment (EarthmanT) [reply cloudify_vsphere/devices/__init__.py]: Actually without `relationships` is wrong. The proper naming convention is 
`cloudify.relationships.[platform].[relationship_name]` and `cloudify.nodes.[platform].[NodeType]`

File: cloudify_vsphere/devices/__init__.py [hunk 3]
```diff
@@ -12,17 +12,63 @@
...
+            net_from_rel.target.instance.runtime_properties.get(
+                NETWORK_NAME, nic_properties.get('name'))
+        switch_distributed = \
+            net_from_rel.target.instance.runtime_properties.get(
+                SWITCH_DISTRIBUTED)
+        if network_name:
+            connected_network = {
+                'name': network_name,
+                'switch_distributed': switch_distributed
+            }
```
[the remark this answers is not in the document]
Comment (EarthmanT) [reply cloudify_vsphere/devices/__init__.py]: I have done something similar.
In the future we should verify to support multiple networks this way.

**Pull request merged**